cmd/ro/roagent/roagent.go

```diff
@@ -0,0 +1,122 @@
+// Package roagent provides ROAgent, which implements the SSH agent interface,
+// forwarding sign requests to a Red October server
+package roagent
+
+import (
+	"bytes"
+	"crypto/rand"
+	"encoding/json"
+	"errors"
+	"io"
+	"log"
+
+	"github.com/cloudflare/redoctober/client"
+	"github.com/cloudflare/redoctober/core"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/agent"
+)
+
+type ROSigner struct {
+	server       *client.RemoteServer
+	pub          ssh.PublicKey
+	encryptedKey []byte
+	user         string
+	pswd         string
+}
+
+func (signer ROSigner) PublicKey() ssh.PublicKey {
+	return signer.pub
+}
+
+func (signer ROSigner) Sign(rand io.Reader, msg []byte) (signature *ssh.Signature, err error) {
+	req := core.SSHSignWithRequest{
+		Name:     signer.user,
+		Password: signer.pswd,
+		Data:     signer.encryptedKey,
+		TBSData:  msg,
+	}
+
+	resp, err := signer.server.SSHSignWith(req)
+	if err != nil {
+		return nil, err
+	}
+	if resp.Status != "ok" {
+		log.Fatal("response status error:", resp.Status)
```

Comment:
This is fixed; though github didn't mark this as a "comment on an outdated diff"